src/promptflow/tests/sdk_cli_test/unittests/test_pf_client.py

```diff
@@ -49,6 +49,16 @@ def test_connection_provider(self):
             client = PFClient()
             assert isinstance(client.connections, LocalAzureConnectionOperations)
 
+        with mock.patch(target) as mocked:
+            mocked.get_instance.return_value.get_connection_provider.return_value = (
+                "azureml:"
+                + RESOURCE_ID_FORMAT.format(
+                    "96aede12-2f73-41cb-b983-6d11a904839b", "promptflow", AZUREML_RESOURCE_PROVIDER, "promptflow-eastus"
+                )
+            )
+            client = PFClient(connection_provider="local")
```

Comment:
have we considered adding a more general way to specify config like below:
PFClient(config = {"connection.provider": "local"}